hotshot/src/lib.rs

```
@@ -243,6 +244,53 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, V: Versions> SystemContext<T
         .await
     }
 
+    async fn load_start_epoch_info(
+        membership: &Arc<RwLock<TYPES::Membership>>,
+        start_epoch_info: &Vec<InitializerEpochInfo<TYPES>>,
+    ) {
+        // If the first element in start_epoch_info does not have a block header,
+        // use that to set_first_epoch instead
+        if let Some(epoch_info) = start_epoch_info.first() {
+            if epoch_info.block_header.is_none() {
+                membership
+                    .write()
+                    .await
+                    .set_first_epoch(epoch_info.epoch, INITIAL_DRB_RESULT);
+            }
+        }
```

I would actually suggest just calling `set_first_epoch` in the loop, as part of the `else` on `if let Some(block_header)`

I think doing it the way it's done here is more likely to result in unexpected behaviour, because we can silently discard parts of the initializer. and I don't think there's much to gain from second-guessing the initializer: if it's misconfigured and results in `set_first_epoch` being called twice, it shouldn't be `load_start_epoch_info`'s job to second-guess it

---

I do think we should log by the way, maybe at `tracing::debug!()`, every call to `set_first_epoch` or `add_drb_result`